client/lib/site/index.js

```diff
@@ -78,7 +86,49 @@ Site.prototype.set = function( attributes ) {
 	}
 
 	return changed;
+};
+
+/**
+ * Replaces all site's attributes. Will remove current attributes if they are not present in the
+ * new ones. Use Site.prototype.set if you want to update one or few attributes instead.
+ *
+ * @param {object} attributes - New attributes for a site.
+ * @returns {boolean} - Returns true if attributes changed, otherwise false.
+ */
+Site.prototype.replace = function( attributes ) {
+	const siteKeys = Object.keys( sitesSchema.patternProperties[ '^\\d+$' ].properties );
+
+	let changed = false;
+
+	// Set attrs which will get computed and will mean the same before doing equality checks
+	// so the 'change' event is not triggered. See Site.prototype.updateComputedAttributes.
+	if ( attributes.hasOwnProperty( 'options' ) ) {
+		let options = attributes.options;
+
+		if ( ! options.hasOwnProperty( 'default_post_format' ) || options.default_post_format === '0' ) {
```

Comment:
We should match the what `updatedComputedAttribtues` is doing. ie `! this.options.default_post_format || this.options.default_post_format === '0'`